### PR TITLE
TCVP-1435 JJ Dispute Status Update Validation

### PR DESCRIPTION
**Pull request by karahanb**
# Description

This PR includes the following proposed change(s):

- [TCVP-1435](https://justice.gov.bc.ca/jira/browse/TCVP-1435)
- Added validation rules for updating statuses of JJ Disputes when an admin resolution is submitted.
- Added unit tests for JJ Dispute status update business rules.
- Updated Staff API to handle 405 not allowed exception if the oracle-data-api returns error due to failed JJ Dispute status update. (Updated the OpenAPI spec json)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Does the change impact or break the Docker build?

- [ ] Yes
- [x] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/BaseTestSuite.java [hunk 1]
```diff
@@ -16,10 +17,14 @@ public class BaseTestSuite {
...
 	@Autowired
 	protected DisputeRepository disputeRepository;
+	
+	@Autowired
+	protected JJDisputeRepository jjDisputeRepository;
 
     @BeforeEach
     protected void beforeEach() throws Exception {
     	disputeRepository.deleteAll();
+    	jjDisputeRepository.deleteAll();
```
Comment (KDNeufeld) [inline src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/BaseTestSuite.java:27]: Probably not in this PR, but we should change this to use @Transactional instead.  This way each unit test rolls back changes.  It also allows us to prepopulate the database with specific test records.  With this deleteAll we can't do that.

File: src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/service/JJDisputeServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+	}
+
+	private JJDispute saveDispute(JJDisputeStatus jjDisputeStatus) {
+		JJDispute jjDispute = RandomUtil.createJJDispute();
+		jjDispute.setStatus(jjDisputeStatus);
+
+		return jjDisputeRepository.save(jjDispute);
+	}
+
+}
```
Comment (KDNeufeld) [inline src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/service/JJDisputeServiceTest.java:91]: Well done. Nice tests.

**Review by KDNeufeld — APPROVED** [review]
LGTM

File: src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/BaseTestSuite.java (hunk 1 above)
Comment (karahanb) [reply src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/BaseTestSuite.java:27]: Yes, good call. Noted for another PR when we have time.

File: src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/service/JJDisputeServiceTest.java (hunk 2 above)
Comment (karahanb) [reply src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/service/JJDisputeServiceTest.java:91]: Thanks, it was mostly inspired from your tests for dispute service :)

**Pull request merged**